src/test/java/com/solvd/LoginTest.java

```diff
@@ -2,56 +2,48 @@
 
 import com.solvd.pages.common.LoginPageBase;
 import com.solvd.pages.common.ProductListPageBase;
+import com.solvd.service.LoginCredentials;
 import org.testng.Assert;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
-import java.lang.reflect.Method;
+import static com.solvd.service.LoginCredentials.*;
 
 public class LoginTest extends BaseTest {
-    //TC001
-    @Test (dataProvider = "data-provider")
-    public void testLoginUser(String name, String password) {
+    //TC002, TC010 and TC001
+    @Test (dataProvider = "loginData")
+    public void testLoginDifferentUsers(LoginCredentials credentials) {
         LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
-        ProductListPageBase productPage = loginPage.login(name, password);
-        Assert.assertTrue(productPage.isTitlePresent(), "Failed to display product page title");
-    }
-
-    //TC002
-    @Test (dataProvider = "data-provider")
-    public void testInvalidLogin(String name, String password) {
-        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
-        ProductListPageBase productPage = loginPage.login(name, password);
-        Assert.assertTrue(loginPage.isErrorMessagePresent(), "Failed to display error");
-        Assert.assertEquals(loginPage.getErrorText(), "Username and password do not match any user in this service.", "Wrong error message");
-    }
-
-    //TC010
-    @Test (dataProvider = "data-provider")
-    public void testLockedOutLogin(String name, String password) {
-        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
-        ProductListPageBase productPage = loginPage.login(name, password);
-        Assert.assertTrue(loginPage.isErrorMessagePresent(), "Failed to display error");
-        Assert.assertEquals(loginPage.getErrorText(), "Sorry, this user has been locked out.", "Wrong error message");
+        ProductListPageBase productPage = loginPage.login(credentials.getName(), credentials.getPassword());
+        switch (credentials) {
```

Comment:
Consider avoiding hardcoded branching via switch. It tightly couples the test logic to specific enum values and makes the test harder to scale. A data-driven approach would be cleaner and easier to maintain.

Reply:
These error messages are hardcoded in the test. It's better to encapsulate expected error messages inside the LoginCredentials enum (e.g., as a field) and use credentials.getExpectedErrorMessage() for cleaner logic and reusability.